fn_ioc_parser/fn_ioc_parser/components/function_ioc_parser.py

```diff
@@ -0,0 +1,106 @@
+        self.options = opts.get("fn_ioc_parser", {})
+
+    @function("function_ioc_parser")
+    def _function_ioc_parser_function(self, event, *args, **kwargs):
+        """Function: A function to extract IOC's(Indicators of Compromise) from (text-based Data) Artifact and attachments."""
+        try:
+            # Get the function parameters:
+            ioc_parser_incident_id = kwargs.get("ioc_parser_incident_id")  # number
+            ioc_parser_task_id = kwargs.get("ioc_parser_task_id")  # number
+            ioc_parser_attachment_id = kwargs.get("ioc_parser_attachment_id")  # number
+            ioc_parser_artifact_id = kwargs.get("ioc_parser_artifact_id")  # number
+            ioc_parser_artifact_type = kwargs.get("ioc_parser_artifact_type")  # text
+            ioc_parser_artifact_value = kwargs.get("ioc_parser_artifact_value")  # text
+
+            log = logging.getLogger(__name__)
+            log.info("ioc_parser_incident_id: %s", ioc_parser_incident_id)
+            log.info("ioc_parser_task_id: %s", ioc_parser_task_id)
+            log.info("ioc_parser_attachment_id: %s", ioc_parser_attachment_id)
+            log.info("ioc_parser_artifact_id: %s", ioc_parser_artifact_id)
+            log.info("ioc_parser_artifact_type: %s", ioc_parser_artifact_type)
+            log.info("ioc_parser_artifact_value: %s", ioc_parser_artifact_value)
+
+            yield StatusMessage("Extracting IOCs from given Artifact/attachment data")
+
+            # Initialising the Resilient Rest client for making api calls to Resilient system
+            resilient_client = self.rest_client()
+
+            # Initialising the IOC Parser Helper class
+            IOCHelp_obj = IOCParserHelper()
+
+            if ioc_parser_artifact_id:
+                # A block to parse and download the data from Artifact
+                if ioc_parser_artifact_type.lower().strip() in ['string', 'email subject']:
+                    ioc_parser_data = ioc_parser_artifact_value
+                else:
+                    metadata_uri = IOCHelp_obj.ARTIFACT_META_DATA_URL.format(ioc_parser_incident_id, ioc_parser_artifact_id)
+                    data_uri = IOCHelp_obj.ARTIFACT_DATA_URL.format(ioc_parser_incident_id, ioc_parser_artifact_id)
+
+                    # Getting the Meta data from the Resilient for an artifact
+                    metadata_data = resilient_client.get(metadata_uri)
+                    if metadata_data.get('attachment'):
+
+                        # Getting the file content from Resilient system
+                        attachment_data = resilient_client.get_content(data_uri)
+
+                        attachment_file_name = metadata_data.get('attachment').get('name')
+                        ioc_parser_data = IOCParserHelper.extract_text_from_bytes_data(attachment_file_name, attachment_data)
+                        print(ioc_parser_data)
+                    else:
+                        ioc_parser_data = metadata_data.get('description')
```

Comment:
* So if the Artifact's metadata does not have an attribute 'attachment' we use the Artifact's description to parse IOCs from?
* I am not sure we need to handle this use case. I don't think we would search an Artifacts Description for an IOC.
* We would only search the Value

Reply:
* Still unsure why we are getting the Artifact Description here...